includes/Data/Patterns.php

```diff
@@ -271,18 +253,14 @@ public static function get_theme_step_patterns_from_step( $step, $squash = false
 		$block_patterns          = array();
 		$block_patterns_squashed = '';
 
-		// fetch the selected header menu slug from DB
-		$flow_data        = \get_option( Options::get_option_name( 'flow' ) );
-		$header_menu_slug = explode( '/', $flow_data['data']['partHeader'] )[1];
-		if ( ! empty( $header_menu_slug ) ) {
-			$pattern_slugs = self::replace_header_menu_slug( $pattern_slugs, $header_menu_slug );
-		}
-
 		foreach ( array_keys( $pattern_slugs ) as $pattern_slug ) {
 			if ( true === $pattern_slugs[ $pattern_slug ]['active'] ) {
 				$pattern_name = $active_theme . '/' . $pattern_slug;
+				$pattern      = $block_patterns_registry->get_registered( $pattern_name );
 				if ( $block_patterns_registry->is_registered( $pattern_name ) ) {
-					$pattern = $block_patterns_registry->get_registered( $pattern_name );
+					if ( array_key_exists( 'content', $pattern_slugs[ $pattern_slug ] ) && ! empty( $pattern_slugs[ $pattern_slug ]['content'] ) ) {
+						$pattern['content'] = $pattern_slugs[ $pattern_slug ]['content'];
+					}
 					// if header menu slug contains "split" replace the menu links with dummy links
```

Comment:
did not understand the reason for adding the content key when it is always blank by default